src/com/sap/piper/PiperGoUtils.groovy

```diff
@@ -43,7 +43,11 @@ class PiperGoUtils implements Serializable {
             }
 
         }
+
         utils.stashWithMessage('piper-bin', 'failed to stash piper binary', 'piper')
+
+        def piperGoVersion = steps.sh(returnStdout: true, script: "./piper version")
```

Comment:
shouldn't we rather natively emit this as logging field from within the binary in addition to the lib repo?
https://github.com/SAP/jenkins-library/blob/master/pkg/log/log.go

@CCFenner what do you think of adding it there instead? Maybe only when running in verbose mode, not sure ...

Reply:
I think it's a good idea to have the version also in a log field available next to the origin repository.

Reply:
What it the benefit of having values in some metadata fields which does not change? Currently we have

```
library="https://github.com/SAP/jenkins-library.git" stepName=xsDeploy
```

The library is fix as far as I understand. the stepName depends on the step currently running. Having the git commit hash would be another fixed field. What is the purpose of having fixed fields in the log message metadata? That cannot be used for filtering - since it is the same always. Are there use cases on a broader scope, e.g. collecing in some logging and alerting system (like e.g. splunk) where we might see logs from various sources?.

Just as suggestion: the repo and the commit belongs closely together. What about having both in one field in the same way we can denote pipeline versions: "<libName>@<commitHash>"

Reply:
**having version together with the lib makes sense in my eyes.**

There are many cases where people use another library, a fork, ...
We and also the users need to be able to identify as much of the failure context with just one glance at the log in case of issues.

Having those fields with every message provides several advantages:
* easily identify important information without browsing a complete log file
* being able to have proper log messages when logs are sent to a central logging system (logrus provides this option out of the box. We just need to leverage it if we see the need)
* ...